In [20]:
library(tidyverse)
library(haven)
library(dplyr)
library(scales)
library(tidyr)

In [19]:
my_data <- read.csv("censusdata.csv")
head(my_data)

PPSORT ABOID AGEGRP AGEIMM ATTSCH BFNMEMB BedRm CFInc CFInc_AT CFSTAT ⋯
1 1      6     13      7     1      0       4     30    27       2      ⋯
2 2      6     11      5     1      0       3     18    18       2      ⋯
3 3      1     13     99     1      0       0      7     7       6      ⋯
4 4      6     16     99     1      0       4     15    15       2      ⋯
5 5      6     18     99     1      0       3     13    13       3      ⋯
6 6      2     16     99     1      0       4      1     1       7      ⋯
  WT7      WT8      WT9      WT10      WT11     WT12     WT13     WT14     
1 22.90113 22.90113 22.90113 249.27876 22.90113 22.90113 22.90113  22.90113
2 22.89379 22.89379 22.89379  22.89379 22.89379 22.89379 22.89379  22.89379
3 22.90134 22.90134 22.90134  22.90134 22.90134 22.90134 22.90134  22.90134
4 22.87713 22.87713 22.87713  22.87713 22.87713 22.87713 22.87713  22.87713
5 22.90134 22.90134 22.90134  22.90134 22.90134 22.90134 22.90134  22.90134
6 22.89284 22.89284 22.89284  22.89284 22.89284 22.89284 22.89284 249.18847
  WT15     WT16    
1 22.90113 22.90113
2 22.89379 22.89379
3 22.90134 22.90134
4 22.87713 22.87713
5 22.90134 22.90134
6 22.89284 22.89284

<h2>Summary statsitics </h2>

In [3]:
my_data <- my_data |> select(NOC21)
head(my_data)

NOC21
1 88   
2 14   
3 99   
4 99   
5 99   
6 99

In [6]:
my_data <- my_data |> filter(NOC21 != 99 & NOC21!= 88, na.rm = TRUE)
head(my_data)
     

NOC21
1 14   
2 25   
3  1   
4  3   
5  6   
6 22

In [7]:

total_obs <- my_data |> nrow()
total_obs

[1] 28782

In total, the dataset contains 28,782 observations, providing a large sample size for summarizing patterns across occupations and industries. 

In [8]:
summary_stats <- my_data |> group_by(NOC21) |> 
                 summarise(count = n())
#proportion = count/total_obs, 
                           #std_dev = proportion*(1-proportion)/count)
summary_stats

stats_table <- summary_stats |> summarise(mean = mean(count), standard_deviation = sd(count),
                                         max = max(count))
stats_table

NOC21 count
1   1     349 
2   2    2988 
3   3    1107 
4   4     714 
5   5    1139 
6   6    1373 
7   7    1343 
8   8     775 
9   9     951 
10 10     474 
11 11     715 
12 12    1973 
13 13     879 
14 14     602 
15 15     492 
16 16     368 
17 17     501 
18 18    1047 
19 19    2011 
20 20    2632 
21 21    1973 
22 22    1380 
23 23     182 
24 24    1048 
25 25     584 
26 26    1182

mean standard_deviation max 
1 1107 708.2738           2988

There are 26 distinct NOC21 occupation groups represented in the data, with counts ranging from 182 to 2,988 observations per group.

On average, there are 1,107 observations per NOC21 group with a standard deviation of about 708, suggesting substantial variation in the size of different occupation categories

In [9]:
ai_data <- read.csv("AIusage.csv") |> rename(C1 = Business.characteristics)
head(ai_data)

REF_DATE GEO    DGUID         
1 2025     Canada 2021A000011124
2 2025     Canada 2021A000011124
3 2025     Canada 2021A000011124
4 2025     Canada 2021A000011124
5 2025     Canada 2021A000011124
6 2025     Canada 2021A000011124
  C1                                                                   
1 North American Industry Classification System (NAICS), all industries
2 Agriculture, forestry, fishing and hunting [11]                      
3 Mining, quarrying, and oil and gas extraction [21]                   
4 Construction [23]                                                    
5 Manufacturing [31-33]                                                
6 Wholesale trade [41]                                                 
  Use.of.artificial.intelligence..AI..by.businesses.or.organizations.in.producing.goods.or.delivering.services.over.the.last.12.months
1 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
2 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
3 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
4 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
5 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
6 Yes, business used AI for producing goods or delivering services over the last 12 months                                            
  UOM     UOM_ID SCALAR_FACTOR SCALAR_ID VECTOR      COORDINATE VALUE STATUS
1 Percent 239    units         0         v1671082146 1.1.1      12.2  A     
2 Percent 239    units         0         v1671082366 1.2.1       1.8  A     
3 Percent 239    units         0         v1671082586 1.3.1       5.6  A     
4 Percent 239    units         0         v1671082806 1.4.1       3.6  A     
5 Percent 239    units         0         v1671083026 1.5.1      13.1  B     
6 Percent 239    units         0         v1671083246 1.6.1      10.6  B     
  SYMBOL TERMINATED DECIMALS
1 NA     NA         1       
2 NA     NA         1       
3 NA     NA         1       
4 NA     NA         1       
5 NA     NA         1       
6 NA     NA         1

In [10]:
ai_data <- ai_data %>% select(C1, VALUE) |>
  filter(C1 %in% c('Agriculture, forestry, fishing and hunting [11]',
                   'Mining, quarrying, and oil and gas extraction [21]',
                   'Construction [23]',
                   'Manufacturing [31-33]',
                   'Wholesale trade [41]',
                   'Retail trade [44-45]',
                   'Transportation and warehousing [48-49]',
                   'Information and cultural industries [51]',
                   'Finance and insurance [52]',
                   'Real estate and rental and leasing [53]',
                   'Professional, scientific and technical services [54]',
                   'Administrative and support, waste management and remediation services [56]',
                   'Health care and social assistance [62]',
                   'Arts, entertainment and recreation [71]',
                   'Accommodation and food services [72]',
                   'Other services (except public administration) [81]'))
ai_data

stats_table <- summarise(ai_data, mean = mean(VALUE), sd = sd(VALUE), max = max(VALUE))
stats_table

C1                                                                        
1  Agriculture, forestry, fishing and hunting [11]                           
2  Mining, quarrying, and oil and gas extraction [21]                        
3  Construction [23]                                                         
4  Manufacturing [31-33]                                                     
5  Wholesale trade [41]                                                      
6  Retail trade [44-45]                                                      
7  Transportation and warehousing [48-49]                                    
8  Information and cultural industries [51]                                  
9  Finance and insurance [52]                                                
10 Real estate and rental and leasing [53]                                   
11 Professional, scientific and technical services [54]                      
12 Administrative and support, waste management and remediation services [56]
13 Health care and social assistance [62]                                    
14 Arts, entertainment and recreation [71]                                   
15 Accommodation and food services [72]                                      
16 Other services (except public administration) [81]                        
   VALUE
1   1.8 
2   5.6 
3   3.6 
4  13.1 
5  10.6 
6   6.6 
7   1.8 
8  35.6 
9  30.6 
10 11.8 
11 31.7 
12  9.8 
13 17.4 
14 14.0 
15  1.5 
16  7.3

mean   sd       max 
1 12.675 10.97278 35.6

The highest reported AI usage is <b>35.6%</b>, showing that at least one sector has a relatively high level of AI adoption compared with the overall average of <b>12.7%<b>.

Professional, scientific and technical services <b>(31.7%)</b> and health care and social assistance <b>(17.4%)</b> are among the sectors with above-average AI adoption, whereas accommodation and food services <b>(1.5%)</b> and ‘other services’ <b>(7.3%)</b> lag behind the overall mean.

Overall, the summary statistics show large differences in both the number of observations by occupation (NOC21) and the level of AI adoption by industry, highlighting substantial heterogeneity in how AI is used across the economy.